sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/csv/CSVHeaderChecker.scala

```diff
@@ -67,8 +69,8 @@ class CSVHeaderChecker(
           var (nameInSchema, nameInHeader) = (fieldNames(i), columnNames(i))
           if (!caseSensitive) {
             // scalastyle:off caselocale
-            nameInSchema = nameInSchema.toLowerCase
-            nameInHeader = nameInHeader.toLowerCase
+            nameInSchema = nameInSchema.toLowerCase(Locale.ROOT)
```

Comment:
These may be user-supplied column names so I think it has to be locale-sensitive